components/stacks-network/src/chains_coordinator.rs

```diff
@@ -572,10 +573,17 @@ pub async fn publish_stacking_orders(
     }
 
     let mut transactions = 0;
-    for (i, pox_stacking_order) in devnet_config.pox_stacking_orders.iter_mut().enumerate() {
-        if (pox_stacking_order.start_at_cycle - 1) as u64 != reward_cycle_id {
+    for (i, pox_stacking_order) in devnet_config.pox_stacking_orders.iter().enumerate() {
+        let PoxStackingOrder {
+            duration,
+            start_at_cycle,
+            ..
+        } = pox_stacking_order;
+
+        if (reward_cycle_id as u32 % duration) != (start_at_cycle - 1) {
             continue;
         }
+        let extend_stacking = reward_cycle_id as u32 != start_at_cycle - 1;
```

Comment:
This is waaaaay better than my attempts to see if we've already stacked; I did that in a hurry 😅 

Shouldn't we also check the `auto_extend` flag to see if the user _wants_ us to extend?

Reply:
Yes you right! I added a check below